src/adm.rs

```diff
@@ -135,12 +175,19 @@ impl AdmFilter {
         tile: &mut AdmTile,
         tags: &mut Tags,
     ) -> HandlerResult<()> {
-        check_url(
-            &tile.impression_url,
-            "Impression",
-            &filter.impression_hosts,
-            tags,
-        )
+        let url = &tile.impression_url;
+        let species = "Impression";
+        tags.add_tag("type", species);
+        tags.add_extra("url", &url);
+        let parsed: Url = match url.parse() {
+            Ok(v) => v,
+            Err(e) => {
+                tags.add_extra("parse_error", &e.to_string());
+                return Err(HandlerErrorKind::InvalidHost(species, url.to_string()).into());
+            }
+        };
+        check_url(parsed, species, &filter.impression_hosts)?;
```

Comment:
Sorry, we forgot one more thing: we should require an `id` param on `impression_url`s